### PR TITLE
Added parsing level and collecting coins

### PR DESCRIPTION
**Pull request by DaniloMatic99**
Level is parsed throw file in data, where '*' describe place on screen where coin need to be placed. Also pick up is finished, and counting collected coins.

### CODE REVIEW

File: data/coin/goldCoin.png.data [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+pivot: 0.5, 0.5
+scale: 1, 1
+
+clip: 48, 0, 16, 16
+pivot: 0.5, 0.5
+scale: 1, 1
+
+clip: 64, 0, 16, 16
+pivot: 0.5, 0.5
+scale: 1, 1
```
Comment (M4444) [inline data/coin/goldCoin.png.data]: No new line at the end of file as GitHub also noted.
Integrating #30 and saving the files again should solve this.

File: src/Background.cpp [hunk 2]
```diff
@@ -18,7 +18,7 @@ void BackgroundSystem::on_tick()
 		}	
 		if (!ourGlobal.isGrounded)
 		{
-			if(counter < 8)
+			if(counter < NUMBER_OF_BACKGROUND-2)
```
Comment (M4444) [inline src/Background.cpp]:
```suggestion
			if (counter < NUMBER_OF_BACKGROUND - 2)
```
This change is also applicable to other places and also with `for` loops.

File: src/Item.cpp
Comment (M4444) [inline src/Item.cpp:1]: This file has a mix of tabs and spaces while tabs should be primary.
#30 will help fix the aliment and make this more obvious.

Also some unnecessary empty rows (e.g. line 5, 8, 26, etc).

File: src/Platform.cpp [hunk 3]
```diff
@@ -1,5 +1,5 @@
 #include "Platform.h"
-
+#include <iostream>
```
Comment (M4444) [inline src/Platform.cpp:2]: Keep an empty row below the includes.

**Review by M4444 — CHANGES_REQUESTED** [review]

File: src/PlayerControls.cpp
Comment (M4444) [inline src/PlayerControls.cpp:1]: Could these magic numbers be refactored out?

**Review by pejovicnina — APPROVED** [review]

File: data/coin/goldCoin.png.data (hunk 1 above)
Comment (M4444) [reply data/coin/goldCoin.png.data:23]: This comment still applies.

File: include/Background.h [hunk 4]
```diff
@@ -4,11 +4,10 @@
...
 #include "GameComponents.h"
 
-
 struct BackgroundSystem
-	: public ecs::System
-	, public MutAccessGroupStorage<Background,KeyBindings, Position, Sprite>
-	, public MutAccessUnique<OurGlobalVar>
+    : public ecs::System,
+      public MutAccessGroupStorage<Background, KeyBindings, Position, Sprite>,
+      public MutAccessUnique<OurGlobalVar>
```
Comment (M4444) [inline include/Background.h]: These should also be tabs and commas at the start to make it consistent.
`class AssetModule` in `Assets.h` is a good example.

File: include/Item.h [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+#include "Definitions.h"
+#include "Access.h"
+#include "GameComponents.h"
+
+struct ItemSystem
+    : public ecs::System,
+      public MutAccessGroupStorage<Item, KeyBindings, Position, Sprite, Visibility>,
+      public MutAccessGroupStorage<Player, Position>,
+      public MutAccessUnique<OurGlobalVar>
```
Comment (M4444) [inline include/Item.h]: Same thing as in `Background.h`.

File: include/Platform.h [hunk 6]
```diff
@@ -5,12 +5,12 @@
...
 
 struct PlatformSystem
-	: public ecs::System
-	, public MutAccessGroupStorage<Platform, KeyBindings, Position, Sprite>
-	, public MutAccessGroupStorage<Player,Position>
-	, public MutAccessUnique<OurGlobalVar>
+    : public ecs::System,
+      public MutAccessGroupStorage<Platform, KeyBindings, Position, Sprite>,
+      public MutAccessGroupStorage<Player, Position>,
+      public MutAccessUnique<OurGlobalVar>
```
Comment (M4444) [inline include/Platform.h]: And again here.

File: src/Main.cpp [hunk 7]
```diff
@@ -3,68 +3,101 @@
...
+#include <string>
+
 using namespace core;
 
-struct FallingHero 
-	: public Game
-	, public AllocateUnique<OurGlobalVar>
+struct FallingHero
+    : public Game,
+      public AllocateUnique<OurGlobalVar>
```
Comment (M4444) [inline src/Main.cpp]: Here too.

**Review by M4444 — CHANGES_REQUESTED** [review]

**Review by M4444 — APPROVED** [review]

**Pull request merged**